osu.Game/Tests/Visual/SkinnableTestScene.cs

```diff
@@ -30,11 +29,15 @@ public abstract class SkinnableTestScene : OsuGridTestScene
         protected SkinnableTestScene()
             : base(2, 3)
         {
+            // avoid running silently incorrectly.
+            if (CreateRuleset() == null)
+            {
+                throw new InvalidOperationException(
+                    $"No ruleset provided, override {nameof(CreateRuleset)} to the ruleset belonging to the skinnable content."
+                    + "This is required to add the legacy skin transformer for the content to behave as expected.");
+            }
```

Comment:
would do the same trick as above with a new abstract property that `CreateRuleset()` would delegate to, to avoid failing at runtime and move the failure to compile-time

Reply:
properties shouldn't return new instances afaik, will make it `CreateRulesetForSkinProvider()` instead.